src/ast/mod.rs

```diff
@@ -1080,6 +1090,12 @@ pub enum Action {
     /// Be wary! Only delete entries that are subsumed in some way or
     /// guaranteed to be not useful.
     Delete(Symbol, Vec<Expr>),
+    /// Set an entry to be `unextractable`
+    /// so that it cannot be extracted.
+    /// Note that this cannot be an expr but has to be a symbol and args, because we need to refer to a specific row
+    Unextractable(Symbol, Vec<Expr>),
```

Comment:
I think in typechecking we should disallow calling `Unextractable` on non-datatype tables, since they can't be extracted anyways. That way they get a good error message.